system_updates.py

```
@@ -0,0 +1,715 @@
+    def remove_stale_locks(self) -> Dict[str, Any]:
+        lock_status = self.get_lock_status()
+        if lock_status["processes"] or lock_status["own_operation_running"] or lock_status["held_locks"]:
+            raise RuntimeError("Apt or dpkg is running. Stop or wait for it before removing stale locks.")
+
+        existing = [str(path) for path in APT_LOCK_PATHS if path.exists()]
+        if self.runtime.is_fake:
+            return {"removed": existing or [str(APT_LOCK_PATHS[0])], "message": "Fake mode: stale apt locks removed."}
+        if not existing:
+            return {"removed": [], "message": "No apt lock files exist."}
+
+        subprocess.run(["sudo", "rm", "-f", *existing], check=True, capture_output=True, text=True)
```

Ruff (0.15.11)

[error] `subprocess` call: check for execution of untrusted input

(S603)